manifests/update.pp

```diff
@@ -56,13 +56,25 @@
   } else {
     $_refresh = true
   }
+  $apt_update_successful = @("ASH"/$)
+    try=1
+    while [ $try -le ${apt::_update['tries']} ]; do
+      timeout ${apt::_update['timeout']} ${apt::provider} update && exit 0
+      [ \$try -eq 1 ] && exit \$?
```

Comment:
if you put this into https://www.shellcheck.net/

```sh
#!/bin/sh
    try=1
    while [ $try -le 10 ]; do
      timeout 11 /usr/bin/apt update && exit 0
      [ $try -eq 1 ] && exit $?
      sleep 1
      try=$(( try + 1 ))
    done
```

it finds https://www.shellcheck.net/wiki/SC2319 which I think is a valid mistake here.

Reply:
@kenyon 
You're probably right. Re-coded; shellcheck is happy now.

Reply:
I do not understand what this code try to do… Try at most 10 times, exit immediately with 0 when apt succeed and exit with the last try exit code if it failed all the times? Then:

```sh
    try=1
    res=-1
    while [ $res -ne 0 -a $try -le 10 ]; do
      timeout 11 /usr/bin/apt update
      res=$?
      if [ $res -ne 0 ]; then
        sleep 1
        try=$(( try + 1 ))
      fi
    done
    exit $res
```

And what's the point of the timeout (the `exec` resource already has one)?

Reply:
The point of the timeout is that it's a timeout *per try*, not a cumulative total timeout.
I really should remove the timeout from the exec resource but a bunch of rspec tests explicitly look for it and it's easier to leave it in.

Reply:
The point of the entire code is to do everything that the original exec did, only in the `unless` param, instead of in the main exec.

If the `unless` code succeeds, no need to run the main `exec` (and therefore, no corrective change gets logged).

If the `unless` code fails, we want to explicitly log a failure; that is the only job of the main exec.